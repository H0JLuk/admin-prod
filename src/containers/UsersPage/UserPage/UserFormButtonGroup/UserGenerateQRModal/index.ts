import UserGenerateQRModal from './UserGenerateQRModal';

export default UserGenerateQRModal;
