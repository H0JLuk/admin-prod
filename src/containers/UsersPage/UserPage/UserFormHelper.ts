import { ClientAppDto } from '@types';

export type getUserAppsCheckboxesResult = Record<string, {id: number; label: string; disabled: boolean; checked: boolean;}>;

export const getUserAppsCheckboxes = (availableApps: ClientAppDto[] = [], checkedApps: number[] = []) =>
    availableApps.reduce<getUserAppsCheckboxesResult>(
        (result, app) => ({
            ...result,
            [app.code]: {
                id: app.id,
                label: app.displayName,
                disabled: false,
                checked: checkedApps.includes(app.id),
            },
        }),
        {}
    );
