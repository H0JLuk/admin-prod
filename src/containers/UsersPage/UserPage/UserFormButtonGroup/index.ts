import UserFormButtonGroup from './UserFormButtonGroup';

export { BUTTON } from './UserFormButtonGroup';

export default UserFormButtonGroup;
