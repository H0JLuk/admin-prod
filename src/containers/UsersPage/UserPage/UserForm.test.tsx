import React from 'react';
import { generatePath } from 'react-router-dom';
import { act, cleanup, fireEvent, render, screen } from '@testing-library/react';
import UserForm, { LOGIN_FIELD } from './UserForm';
import { getUserAppsCheckboxes } from './UserFormHelper';
import {
    getCurrUserInteractionsForOtherUser,
    getCommonPermissionsByRole,
} from '@constants/permissions';
import { getUser, unblockUser, resetUser, removeUser, editUser, addUser } from '@apiServices/usersService';
import { getActiveClientApps } from '@apiServices/clientAppService';
import { sleep } from '@setupTests';
import { INFO_USER_BUTTONS } from './UserFormButtonGroup/UserFormButtonGroup';
import { customNotifications } from '@utils/notifications';
import {
    userTestData,
    clientAppListTestResponse,
    searchSalePointTestData,
    testSalePoint,
} from '@testConstants';
import { getResultsByTextAndType, createSearchDataAndPassLocation } from '@components/AutoComplete/AutocompleteLocationAndSalePoint/AutocompleteHelper';
import { confirmModal } from '@utils/utils';
import { getSalePointByText } from '@apiServices/salePointService';
import { BUTTON_TEXT } from '@constants/common';

const mockSalePointSelect = {
    ...searchSalePointTestData[0],
    type: {
        ...searchSalePointTestData[0].type,
        kind: 'INTERNAL',
    },
};
jest.mock(
    '@components/AutoComplete/AutocompleteLocationAndSalePoint',
    () => ({ onLocationChange, onSalePointChange, error, locationId }: any) => (
        <>
            {error.salePoint}
            <span data-testid="locationId">{locationId}</span>
            <button data-testid="location" onClick={() => onLocationChange({ id: 2, name: 'testName' })}>location</button>
            <button data-testid="salePoint" onClick={() => onSalePointChange(mockSalePointSelect)}>salePoint</button>
        </>
    ),
);

jest.mock('antd', () => ({
    ...jest.requireActual('antd'),
    Select({ onChange }: any) {
        return (
            <div data-testid="Select">
                <button data-testid="changeRoleToPartner" onClick={() => onChange('Partner')}>PartnerRole</button>
                <button data-testid="changeRoleToUser" onClick={() => onChange('User')}>UserRole</button>
            </div>
        );
    },
}));

jest.mock('@components/Checkboxes', () => ({ checkboxesData, onChangeAll, onChange }: any) => (
    <div data-testid="checkboxes">
        <span data-testid="checkboxesData">{JSON.stringify(checkboxesData)}</span>
        <button data-testid="changeAllCheckboxes" onClick={(e: any) => onChangeAll(e.target.testValue)}>changeAllCheckboxes</button>
        <button data-testid="changeOneCheckbox" onClick={(e: any) => onChange(e.target.testValue, 'digital-village')}>changeOneCheckbox</button>
    </div>
));

jest.mock(
    '@components/AutoComplete',
    () => ({ value, error, onSelect, componentMethods }: any) => {
        componentMethods.current = { clearState: () => '' };
        return (
            <div data-testid="autocompletePartner">
                <div data-testid="autocompletePartnerValue">{JSON.stringify(value)}</div>
                <button data-testid="partnerSelect" onClick={() => onSelect({ personalNumber: 'testPN' })}>partnerSelect</button>
                <span data-testid="partnerError">{error}</span>
            </div>
        );
    },
);

jest.mock('@apiServices/salePointService', () => ({
    getSalePointByText: jest.fn(),
}));

jest.mock('@components/AutoComplete/AutocompleteLocationAndSalePoint/AutocompleteHelper', () => ({
    getResultsByTextAndType: jest.fn(),
    createSearchDataAndPassLocation: jest.fn(),
}));

jest.mock('@apiServices/clientAppService', () => ({
    getActiveClientApps: jest.fn(),
}));

jest.mock('@utils/utils', () => ({
    ...jest.requireActual('@utils/utils'),
    confirmModal: jest.fn(),
}));

jest.mock('@constants/permissions', () => {
    const originalModule = jest.requireActual('@constants/permissions');

    return {
        __esModule: true,
        ...originalModule,
        getCurrUserInteractionsForOtherUser: jest.fn(),
        getCommonPermissionsByRole: jest.fn(),
    };
});

jest.mock('@apiServices/usersService', () => ({
    getUser: jest.fn(),
    unblockUser: jest.fn(),
    resetUser: jest.fn(),
    removeUser: jest.fn(),
    editUser: jest.fn(),
    addUser: jest.fn(),
}));

jest.mock('./UserFormHelper', () => {
    const originalModule = jest.requireActual('./UserFormHelper');

    return {
        __esModule: true,
        ...originalModule,
        getUserAppsCheckboxes: jest.fn(),
    };
});

const mockHistoryPush = jest.fn();

let mockErrorParams = false;
jest.mock('react-router-dom', () => ({
    useParams: () => !mockErrorParams ? ({ userId: 1 }) : ({ userId: 'testId' }),
    useHistory: () => ({
        push: mockHistoryPush,
        replace: jest.fn(),
    }),
    generatePath: jest.fn(),
}));

const TEST_PASSWORD = 'TEST_PASSWORD_123';

const SearchData = {
    results: [testSalePoint],
    value: 'Московская область, Москва',
};

const LocationData = searchSalePointTestData[0].location;

const TEST_PASSWORDS = {
    newPassword: TEST_PASSWORD,
    generatedPassword: TEST_PASSWORD,
};

const TEST_USER_DATA = {
    FALSE_LOGIN: '-=-',
    LOGIN: '123456',
    SALE_POINT: '1232',
};

const CASE_TYPES = {
    NEW: 'new',
    EDIT: 'edit',
    INFO: 'info',
    NONE: '',
};

const TEST_PROPS = {
    matchPath: '/TEST_PATH',
};

const TEST_INTERACTIONS = {
    canSelectUserInTable: true,
    deleteUser: true,
    editUser: true,
    resetUserPassword: true,
    unlockUser: true,
};

const TEST_COMMON_PERMISSIONS = {
    canSetUserRole: true,
    canSetUserPartner: true,
};

const TEST_INTERACTIONS_FALSE = {
    canSelectUserInTable: false,
    deleteUser: false,
    editUser: false,
    resetUserPassword: false,
    unlockUser: false,
};


const TEST_CHECKBOX = {
    'digital-village': {
        checked: false,
        disabled: false,
        id: 9,
        label: 'Партнеры',
    },
};

describe('<UserForm /> test', () => {
    const Component = (type = CASE_TYPES.INFO) => render(
        <UserForm {...TEST_PROPS} type={type} />,
    );

    beforeEach(() => {
        (getActiveClientApps as jest.Mock).mockResolvedValue(clientAppListTestResponse.list);
        (getUser as jest.Mock).mockResolvedValue(userTestData);
        (getCurrUserInteractionsForOtherUser as jest.Mock).mockReturnValue(TEST_INTERACTIONS);
        (getCommonPermissionsByRole as jest.Mock).mockReturnValue(TEST_COMMON_PERMISSIONS);
        (getUserAppsCheckboxes as jest.Mock).mockReturnValue(TEST_CHECKBOX);
        (getSalePointByText as jest.Mock).mockReturnValue({ ...searchSalePointTestData[0], id: userTestData.salePointId });
        cleanup();
        document.body.innerHTML = '';
    });

    it('should be mount snap with useEffect', async () => {
        const ComponentInfo = Component();
        await sleep();
        expect(getActiveClientApps).toBeCalledTimes(1);
        expect(getUser).toBeCalledTimes(1);
        expect(getCurrUserInteractionsForOtherUser).toBeCalledTimes(1);
        expect(getUserAppsCheckboxes).toBeCalledTimes(1);
        await sleep();
        expect(ComponentInfo).toMatchSnapshot();
    });

    it('should find buttons and call reset password function', async () => {
        (resetUser as jest.Mock).mockResolvedValue(TEST_PASSWORDS);
        await act(async () => {
            Component();
        });

        expect(screen.getByText(BUTTON_TEXT.RESET_PASS)).toBeTruthy();
        expect(screen.getByText(BUTTON_TEXT.EDIT)).toBeTruthy();
        expect(screen.getByText(BUTTON_TEXT.DELETE)).toBeTruthy();

        await act(async () => {
            fireEvent.click(screen.getByText(/Сбросить/i));
        });

        expect(resetUser).toBeCalledTimes(1);
        expect(screen.getByText(/TEST_PASSWORD_123/i)).toBeTruthy();
    });

    it('should find buttons and call reset password function with catch', async () => {
        (resetUser as jest.Mock).mockRejectedValue(new Error('Error'));
        customNotifications.error = jest.fn();

        await act(async () => {
            Component();
        });

        expect(screen.getByText(BUTTON_TEXT.RESET_PASS)).toBeTruthy();
        expect(screen.getByText(BUTTON_TEXT.EDIT)).toBeTruthy();
        expect(screen.getByText(BUTTON_TEXT.DELETE)).toBeTruthy();

        await act(async () => {
            fireEvent.click(screen.getByText(/Сбросить/i));
        });

        expect(resetUser).toBeCalledTimes(1);
        expect(customNotifications.error).toBeCalledTimes(1);
    });

    it('should call `generatePath` function', async () => {
        Component();
        await sleep();
        expect(screen.getByText(BUTTON_TEXT.EDIT)).toBeTruthy();
        fireEvent.click(screen.getByText(BUTTON_TEXT.EDIT));
        expect(generatePath).toBeCalledTimes(1);
    });

    it('should call confirmModal and delete function', async () => {
        Component();
        await sleep();
        expect(screen.getByText(BUTTON_TEXT.DELETE)).toBeTruthy();
        fireEvent.click(screen.getByText(BUTTON_TEXT.DELETE));
        await sleep();
        expect(confirmModal).toBeCalledTimes(1);

        await act(async () => {
            await (confirmModal as jest.Mock).mock.calls[0][0].onOk();
        });
        expect(removeUser).toBeCalledTimes(1);
    });

    it('should call delete function with catch', async () => {
        (removeUser as jest.Mock).mockRejectedValue(new Error('Error'));
        customNotifications.error = jest.fn();
        Component();
        await sleep();
        expect(screen.getByText(BUTTON_TEXT.DELETE)).toBeTruthy();
        fireEvent.click(screen.getByText(BUTTON_TEXT.DELETE));
        expect(confirmModal).toBeCalledTimes(1);

        await act(async () => {
            await (confirmModal as jest.Mock).mock.calls[0][0].onOk();
        });
        await sleep();
        expect(removeUser).toBeCalledTimes(1);
        expect(customNotifications.error).toBeCalledTimes(1);
    });

    it('should call catch in onDelete function', async () => {
        (getUser as jest.Mock).mockResolvedValue({ ...userTestData, id: null });
        Component();
        await sleep();
        expect(screen.getByText(BUTTON_TEXT.DELETE)).toBeTruthy();
        fireEvent.click(screen.getByText(BUTTON_TEXT.DELETE));
        expect(confirmModal).toBeCalledTimes(1);

        await act(async () => {
            await (confirmModal as jest.Mock).mock.calls[0][0].onOk();
        });
        expect(removeUser).toBeCalledTimes(1);
    });

    it('should call submit function', async () => {
        (getSalePointByText as jest.Mock).mockReturnValue({
            ...searchSalePointTestData[0],
            id: userTestData.salePointId,
            type: {
                ...searchSalePointTestData[0].type,
                kind: 'INTERNAL',
            },
        });
        (editUser as jest.Mock).mockResolvedValue(TEST_PASSWORDS);
        Component(CASE_TYPES.EDIT);
        await sleep();

        expect(screen.getByText(BUTTON_TEXT.SAVE)).toBeTruthy();
        fireEvent.click(screen.getByText(BUTTON_TEXT.SAVE));
        await sleep();

        expect(editUser).toBeCalledTimes(1);
        expect(generatePath).toBeCalledTimes(1);
    });

    it('should call submit function with catch', async () => {
        (getSalePointByText as jest.Mock).mockReturnValue({
            ...searchSalePointTestData[0],
            id: userTestData.salePointId,
            type: {
                ...searchSalePointTestData[0].type,
                kind: 'INTERNAL',
            },
        });
        (editUser as jest.Mock).mockRejectedValue(new Error('Error'));
        Component(CASE_TYPES.EDIT);
        await sleep();

        expect(screen.getByText(BUTTON_TEXT.SAVE)).toBeTruthy();
        fireEvent.click(screen.getByText(BUTTON_TEXT.SAVE));
        await sleep();
        expect(editUser).toBeCalledTimes(1);
        expect(generatePath).toBeCalledTimes(0);
    });

    it('should call check validate if data submit', async () => {
        await act(async () => {
            Component(CASE_TYPES.NEW);
        });

        await act(async () => {
            fireEvent.click(screen.getByText(BUTTON_TEXT.ADD));
        });
        expect(screen.getByText(/Укажите/i)).toBeTruthy();

        await act(async () => {
            fireEvent.change(screen.getByPlaceholderText(LOGIN_FIELD.placeholder), { target: { value: TEST_USER_DATA.FALSE_LOGIN } });
        });
        await act(async () => {
            fireEvent.click(screen.getByText(BUTTON_TEXT.ADD));
        });
        expect(screen.getByText(/латинские/i)).toBeTruthy();

        await act(async () => {
            fireEvent.change(screen.getByPlaceholderText(LOGIN_FIELD.placeholder), { target: { value: TEST_USER_DATA.LOGIN } });
        });
        await act(async () => {
            fireEvent.click(screen.getByText(BUTTON_TEXT.ADD));
        });
        expect(screen.getByText(/продажи/i)).toBeTruthy();
    });

    it('should call submit function with new User', async () => {
        (getResultsByTextAndType as jest.Mock).mockResolvedValue(searchSalePointTestData);
        (addUser as jest.Mock).mockResolvedValue({ generatedPassword: TEST_PASSWORD });
        (createSearchDataAndPassLocation as jest.Mock).mockReturnValue({ location: LocationData, searchLocation: SearchData });
        customNotifications.success = jest.fn();

        await act(async () => {
            Component(CASE_TYPES.NEW);
        });

        expect(screen.getByTestId('locationId').innerHTML).toBe('');

        await act(async () => {
            fireEvent.change(screen.getByPlaceholderText(LOGIN_FIELD.placeholder), { target: { value: TEST_USER_DATA.LOGIN } });
        });
        await act(async () => {
            fireEvent.click(screen.getByTestId('salePoint'));
        });
        await act(async () => {
            fireEvent.click(screen.getByTestId('location'));
        });

        expect(screen.getByTestId('locationId').innerHTML).toBe('2');

        fireEvent.click(screen.getByText(BUTTON_TEXT.ADD));
        expect(addUser).toBeCalledTimes(1);
        await sleep();
        expect(customNotifications.success).toBeCalledTimes(1);
        expect(mockHistoryPush).toBeCalledTimes(1);
    });

    it('should call checkboxes handlers', async () => {
        Component(CASE_TYPES.EDIT);
        await sleep();
        const checkboxesData = JSON.parse(screen.getByTestId('checkboxesData').innerHTML);
        expect(checkboxesData['digital-village'].checked).toBe(false);

        fireEvent.click(
            screen.getByTestId('changeAllCheckboxes'),
            { target: { testValue: true } },
        );
        await sleep();
        expect(
            JSON.parse(
                screen.getByTestId('checkboxesData').innerHTML,
            )['digital-village'].checked,
        ).toBe(true);

        fireEvent.click(
            screen.getByTestId('changeOneCheckbox'),
            { target: { testValue: false } },
        );
        await sleep();
        expect(
            JSON.parse(
                screen.getByTestId('checkboxesData').innerHTML,
            )['digital-village'].checked,
        ).toBe(false);
    });

    it('should call cancel function', async () => {
        Component(CASE_TYPES.EDIT);
        await sleep();
        expect(screen.getByText(BUTTON_TEXT.CANCEL)).toBeTruthy();
        fireEvent.click(screen.getByText(BUTTON_TEXT.CANCEL));
        expect(generatePath).toBeCalledTimes(1);
    });

    it('should call cancel function with redirect', async () => {
        Component(CASE_TYPES.NEW);
        await sleep();
        expect(screen.getByText(BUTTON_TEXT.CANCEL)).toBeTruthy();
        fireEvent.click(screen.getByText(BUTTON_TEXT.CANCEL));
        expect(mockHistoryPush).toBeCalledTimes(1);
    });

    it('should call history replace in get user data', async () => {
        (getCurrUserInteractionsForOtherUser as jest.Mock).mockReturnValue(TEST_INTERACTIONS_FALSE);
        Component(CASE_TYPES.EDIT);
        await sleep();
        expect(generatePath).toBeCalledTimes(1);
    });

    it('should call reset function with blocked user', async () => {
        (getUser as jest.Mock).mockResolvedValue({ ...userTestData, tmpBlocked: true });
        await act(async () => {
            Component();
        });

        await act(async () => {
            fireEvent.click(screen.getByText(INFO_USER_BUTTONS.UNBLOCK));
        });

        expect(unblockUser).toBeCalledTimes(1);
    });

    it('should call catch in get user data', async () => {
        jest.spyOn(console, 'warn').mockImplementation(() => '');
        (getUser as jest.Mock).mockRejectedValue(new Error('Error'));
        Component();
        await sleep();
        expect(mockHistoryPush).toBeCalledTimes(1);
        expect(mockHistoryPush).toBeCalledWith(TEST_PROPS.matchPath);
        expect(console.warn).toBeCalled();
    });

    it('should redirect on first useEffect if userId is wrong', async () => {
        mockErrorParams = true;
        Component();
        await sleep();

        expect(mockHistoryPush).toBeCalledTimes(1);
        expect(mockHistoryPush).toBeCalledWith(TEST_PROPS.matchPath);
        mockErrorParams = false;
    });

    it('should change partner field', async () => {
        Component('edit');
        await sleep();

        await act(async () => {
            fireEvent.click(screen.getByTestId('partnerSelect'));
        });
        expect(screen.getByTestId('autocompletePartnerValue').innerHTML).toBe(JSON.stringify('testPN'));
    });

    it('should change role', async () => {
        Component('new');
        await sleep();

        const login = screen.getByPlaceholderText('Табельный номер') as HTMLInputElement;
        login.value = 'testValue';

        await act(async () => {
            fireEvent.click(screen.getByTestId('partnerSelect'));
        });
        expect(screen.getByTestId('autocompletePartnerValue').innerHTML).toBe(JSON.stringify('testPN'));

        act(() => {
            fireEvent.click(screen.getByTestId('changeRoleToPartner'));
        });

        expect((screen.getByPlaceholderText('Табельный номер') as HTMLInputElement).value).toBe('');
        expect(screen.getByTestId('autocompletePartnerValue').innerHTML).toBe(JSON.stringify(null));
    });
});
