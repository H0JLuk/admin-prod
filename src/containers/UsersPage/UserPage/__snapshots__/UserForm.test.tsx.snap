// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserForm /> test should be mount snap with useEffect 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="backButton"
        >
          <span
            aria-label="left"
            class="anticon anticon-left"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="left"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
              />
            </svg>
          </span>
           
           Назад
        </div>
        <div
          class="avatar"
        >
          <span
            aria-label="user"
            class="anticon anticon-user ant-dropdown-trigger dropdown"
            role="img"
            style="font-size: 22px; color: rgb(255, 255, 255); background-color: rgb(193, 200, 209); border-radius: 50px; width: 32px; height: 32px; padding: 3px;"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              data-icon="user"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="pageWrapper"
        >
          <div
            class="pageTitle"
          >
            Пользователь 111111
            <div
              class="userStatus"
            >
              <div
                class="statusOn"
              />
            </div>
            <div
              class="status"
            >
              Временный пароль
            </div>
          </div>
          <form
            class="ant-form ant-form-horizontal formStyle"
          >
            <div
              class="ant-row ant-row-space-between formWrapper"
              style="margin-top: -7.5px; margin-bottom: -7.5px;"
            >
              <div
                class="loginContainer"
              >
                <label
                  class=""
                  for="login"
                >
                  Логин пользователя
                </label>
                <div
                  class="noEditField"
                  id="idInput"
                >
                  111111
                </div>
              </div>
              <div
                class="labelCol"
              >
                <label>
                  Локация
                </label>
                <div
                  class="noEditField"
                >
                  Калининград
                </div>
              </div>
              <div
                class="labelCol"
              >
                <label>
                  Точка продажи
                </label>
                <div
                  class="noEditField"
                >
                  055_8626_1232
                </div>
              </div>
              <div
                class="ant-col ant-col-8 roleBlock"
                style="padding-top: 7.5px; padding-bottom: 7.5px;"
              >
                <label>
                  Способ авторизации
                </label>
                <div
                  class="noEditField"
                >
                  Логин/Пароль
                </div>
              </div>
              <div
                class="ant-col ant-col-8 labelCol"
                style="padding-top: 7.5px; padding-bottom: 7.5px;"
              >
                <label
                  for="role"
                >
                  Роль
                </label>
                <div
                  class="noEditField"
                >
                  Пользователь
                </div>
              </div>
              <div
                class="ant-col ant-col-8 labelCol"
                style="padding-top: 7.5px; padding-bottom: 7.5px;"
              >
                <label
                  class=""
                >
                  Партнёр
                </label>
                <div
                  class="noEditField"
                >
                  <i>
                    Нет партнёра
                  </i>
                </div>
              </div>
            </div>
          </form>
        </div>
        <div
          class="checkboxesWrapper"
        >
          <h4
            class="title"
          >
            Доступные приложения
          </h4>
          <div
            class="checkBoxesCol"
          >
            <div
              data-testid="checkboxes"
            >
              <span
                data-testid="checkboxesData"
              >
                {"digital-village":{"checked":false,"disabled":false,"id":9,"label":"Партнеры"}}
              </span>
              <button
                data-testid="changeAllCheckboxes"
              >
                changeAllCheckboxes
              </button>
              <button
                data-testid="changeOneCheckbox"
              >
                changeOneCheckbox
              </button>
            </div>
          </div>
        </div>
        <div
          class="buttonsContainer"
        >
          <button
            class="ant-btn ant-btn-primary"
            type="button"
          >
            <span>
              Сбросить пароль
            </span>
          </button>
          <button
            class="ant-btn ant-btn-primary"
            type="button"
          >
            <span>
              Редактировать
            </span>
          </button>
          <button
            class="ant-btn ant-btn-primary ant-btn-dangerous"
            type="button"
          >
            <span>
              Удалить
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="backButton"
      >
        <span
          aria-label="left"
          class="anticon anticon-left"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="left"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
            />
          </svg>
        </span>
         
         Назад
      </div>
      <div
        class="avatar"
      >
        <span
          aria-label="user"
          class="anticon anticon-user ant-dropdown-trigger dropdown"
          role="img"
          style="font-size: 22px; color: rgb(255, 255, 255); background-color: rgb(193, 200, 209); border-radius: 50px; width: 32px; height: 32px; padding: 3px;"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            data-icon="user"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="pageWrapper"
      >
        <div
          class="pageTitle"
        >
          Пользователь 111111
          <div
            class="userStatus"
          >
            <div
              class="statusOn"
            />
          </div>
          <div
            class="status"
          >
            Временный пароль
          </div>
        </div>
        <form
          class="ant-form ant-form-horizontal formStyle"
        >
          <div
            class="ant-row ant-row-space-between formWrapper"
            style="margin-top: -7.5px; margin-bottom: -7.5px;"
          >
            <div
              class="loginContainer"
            >
              <label
                class=""
                for="login"
              >
                Логин пользователя
              </label>
              <div
                class="noEditField"
                id="idInput"
              >
                111111
              </div>
            </div>
            <div
              class="labelCol"
            >
              <label>
                Локация
              </label>
              <div
                class="noEditField"
              >
                Калининград
              </div>
            </div>
            <div
              class="labelCol"
            >
              <label>
                Точка продажи
              </label>
              <div
                class="noEditField"
              >
                055_8626_1232
              </div>
            </div>
            <div
              class="ant-col ant-col-8 roleBlock"
              style="padding-top: 7.5px; padding-bottom: 7.5px;"
            >
              <label>
                Способ авторизации
              </label>
              <div
                class="noEditField"
              >
                Логин/Пароль
              </div>
            </div>
            <div
              class="ant-col ant-col-8 labelCol"
              style="padding-top: 7.5px; padding-bottom: 7.5px;"
            >
              <label
                for="role"
              >
                Роль
              </label>
              <div
                class="noEditField"
              >
                Пользователь
              </div>
            </div>
            <div
              class="ant-col ant-col-8 labelCol"
              style="padding-top: 7.5px; padding-bottom: 7.5px;"
            >
              <label
                class=""
              >
                Партнёр
              </label>
              <div
                class="noEditField"
              >
                <i>
                  Нет партнёра
                </i>
              </div>
            </div>
          </div>
        </form>
      </div>
      <div
        class="checkboxesWrapper"
      >
        <h4
          class="title"
        >
          Доступные приложения
        </h4>
        <div
          class="checkBoxesCol"
        >
          <div
            data-testid="checkboxes"
          >
            <span
              data-testid="checkboxesData"
            >
              {"digital-village":{"checked":false,"disabled":false,"id":9,"label":"Партнеры"}}
            </span>
            <button
              data-testid="changeAllCheckboxes"
            >
              changeAllCheckboxes
            </button>
            <button
              data-testid="changeOneCheckbox"
            >
              changeOneCheckbox
            </button>
          </div>
        </div>
      </div>
      <div
        class="buttonsContainer"
      >
        <button
          class="ant-btn ant-btn-primary"
          type="button"
        >
          <span>
            Сбросить пароль
          </span>
        </button>
        <button
          class="ant-btn ant-btn-primary"
          type="button"
        >
          <span>
            Редактировать
          </span>
        </button>
        <button
          class="ant-btn ant-btn-primary ant-btn-dangerous"
          type="button"
        >
          <span>
            Удалить
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
